packages/docs/docs/components/code/index.mdx

```diff
@@ -0,0 +1,791 @@
+  const code = createRef<Code>();
+
+  view.add(
+    <Code
+      ref={code}
+      fontSize={28}
+      fontFamily={'JetBrains Mono'}
+      offsetX={-1}
+      x={-400}
+      code={`\
+function example() {
+  console.log('Hello!');
+}`}
+    />,
+  );
+
+  yield* code().code.replace(
+    // find the range of "example" and replace it with "greet"
+    code().findFirstRange('example'),
+    'greet',
+    0.6,
+  );
+
+  yield* waitFor(0.6);
+});
+```
+
+## Code Selection
+
+The [`selection`](/api/2d/code/CodeSignal#selection) property can be used to
+visually distinguish a part of the code snippet. The selection is specified
+using an individual [`code range`](#code-ranges) or an array of ranges:
+
+```tsx editor
+import {makeScene2D, Code, lines} from '@motion-canvas/2d';
+import {createRef, DEFAULT, waitFor} from '@motion-canvas/core';
+
+export default makeScene2D(function* (view) {
+  const code = createRef<Code>();
+
+  view.add(
+    <Code
+      ref={code}
+      fontSize={28}
+      fontFamily={'JetBrains Mono'}
+      offsetX={-1}
+      x={-400}
+      code={`\
+function hello() {
+  console.log('Hello');
+}`}
+    />,
+  );
+
+  // select all instances of "hello" (case-insensitive)
+  yield* code().selection(code().findAllRanges(/hello/gi), 0.6);
+  yield* waitFor(0.3);
+
+  // select line 1
+  yield* code().selection(lines(1), 0.6);
+  yield* waitFor(0.3);
+
+  // reset the selection
+  yield* code().selection(DEFAULT, 0.6);
+  yield* waitFor(0.3);
+});
+```
+
+## Querying Positions and Sizes
+
+[`getPointBBox`](/api/2d/code/Code#getPointBBox) and
+[`getSelectionBBox`](/api/2d/code/Code#getSelectionBBox) can be used to retrieve
+the position and size of a specific character or a range of characters,
+respectively. The returned value is a [`bounding box`](/api/core/types/BBox) in
+the local space of the `Code` node.
+
+In the following example, we use `getSelectionBbox` to draw a rectangle around
+the word `log`:
+
+```tsx editor
+import {Code, Rect, makeScene2D} from '@motion-canvas/2d';
+import {createRef, createSignal} from '@motion-canvas/core';
+
+export default makeScene2D(function* (view) {
+  const code = createRef<Code>();
+
+  view.add(
+    <Code
+      ref={code}
+      fontSize={28}
+      fontFamily={'JetBrains Mono'}
+      offsetX={-1}
+      x={-400}
+      code={`\
+function hello() {
+  console.log('Hello');
+}`}
+    />,
+  );
+
+  const range = createSignal(() => {
+    const range = code().findFirstRange('log');
+    const bboxes = code().getSelectionBBox(range);
+    // "getSelectionBBox" returns an array of bboxes,
+    // one for each line in the range. We'll just use
+    // the first one for this example.
+    const first = bboxes[0];
+    return first.expand([4, 8]);
+  });
+
+  code().add(
+    <Rect
+      offset={-1}
+      position={range().position}
+      size={range().size}
+      lineWidth={4}
+      stroke={'white'}
+      radius={8}
+    />,
+  );
+});
+```
+
+## Multiple Languages
```

Comment:
This (or a section after this) might be a good place to document how to use a Lezer dialect, too:
```tsx
LezerHighlighter.registerParser(
  javascriptParser.configure({ dialect: "jsx" }),
  "jsx"
);
LezerHighlighter.registerParser(
  javascriptParser.configure({ dialect: "ts" }),
  "ts"
);
```

Reply:
True, I'll add this in the next PR to not bloat up this one.